source/adios2/toolkit/format/bp5/BP5Deserializer.cpp

```diff
@@ -2127,16 +2127,15 @@ Dims *BP5Deserializer::VarShape(const VariableBase &Var,
             (MetaArrayRec *)GetMetadataBase(VarRec, AbsStep, WriterRank);
         if (writer_meta_base && writer_meta_base->Shape)
         {
-            Dims *Shape = new Dims();
-            Shape->reserve(writer_meta_base->Dims);
+            Shape.resize(writer_meta_base->Dims);
```

Comment:
If Shape has a bigger size than the writer_meta_base, resize will remove the values but not update the size. Would this be a problem? This is not the case in any of the test so I assume no, but just want to point this out.

Reply:
I wouldn't think it would be a problem.  In all the current uses, Shape passed in by reference is an fresh Dims value with default size 0.  So, I think no worries...